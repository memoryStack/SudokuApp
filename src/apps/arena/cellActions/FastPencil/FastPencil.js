import React from 'react'

import PropTypes from 'prop-types'

import _noop from '@lodash/noop'

import { PencilIcon } from '@resources/svgIcons/pencil'

import Text from '@ui/atoms/Text'

import { Touchable } from '../../../components/Touchable'

import { styles, INACTIVE_ICON_FILL } from '../style'

const FastPencil = ({ iconBoxSize, onClick, ...rest }) => (
    <Touchable style={styles.actionContainer} onPress={onClick} {...rest}>
        <PencilIcon iconBoxSize={iconBoxSize} fill={INACTIVE_ICON_FILL} />
        <Text style={styles.actionText}>Fast Pencil</Text>
    </Touchable>
)

export default React.memo(FastPencil)

FastPencil.propTypes = {
    iconBoxSize: PropTypes.number,
    onClick: PropTypes.func,
}

FastPencil.defaultProps = {
    iconBoxSize: 40,
    onClick: _noop,
}
