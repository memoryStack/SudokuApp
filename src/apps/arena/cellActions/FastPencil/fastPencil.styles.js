import { StyleSheet } from 'react-native'

import _get from '@lodash/get'

export const getStyles = ({}, theme) => {
    return StyleSheet.create({

    })
}
