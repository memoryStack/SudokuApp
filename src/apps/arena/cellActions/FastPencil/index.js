export { default } from './FastPencil'
export * from './fastPencil.constants'
