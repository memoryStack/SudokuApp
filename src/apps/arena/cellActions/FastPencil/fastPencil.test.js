import * as React from 'react'

import {
    fireEvent, render, screen,
} from '@utils/testing/testingLibrary'

import FastPencil from './FastPencil'

test('snapshot for  FastPencil', () => {
    const tree = render(<FastPencil iconBoxSize={50} />).toJSON()
    expect(tree).toMatchSnapshot()
})

describe('FastPencil', () => {
    test('will call handler on click', () => {
        const onClick = jest.fn()
        render(<FastPencil iconBoxSize={50} onClick={onClick} />)

        fireEvent.press(screen.getByText('Fast Pencil'))

        expect(onClick).toHaveBeenCalledTimes(1)
    })

    test('if disabled, will not call handler on click', () => {
        const onClick = jest.fn()
        render(<FastPencil iconBoxSize={50} onClick={onClick} disabled />)

        fireEvent.press(screen.getByText('Fast Pencil'))

        expect(onClick).not.toHaveBeenCalled()
    })
})
