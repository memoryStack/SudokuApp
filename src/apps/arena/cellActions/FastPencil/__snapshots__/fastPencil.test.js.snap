// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for  FastPencil 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={null}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={50}
    bbWidth={46}
    focusable={false}
    height={50}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 50,
          "width": 46,
        },
      ]
    }
    vbHeight={512}
    vbWidth={512}
    width={46}
  >
    <RNSVGGroup>
      <RNSVGPath
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill={4286545791}
        propList={
          Array [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Object {
        "color": "rgb(127, 127, 127)",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "includeFontPadding": false,
        "letterSpacing": 0.5,
        "lineHeight": 24,
        "marginTop": 8,
        "textAlignVertical": "center",
      }
    }
  >
    Fast Pencil
  </Text>
</View>
`;
